crates/telio-traversal/src/session_keeper.rs

```diff
@@ -105,13 +139,36 @@ impl SessionKeeperTrait for SessionKeeper {
                 interval,
                 Arc::new(move |c| {
                     Box::pin(async move {
-                        telio_log_debug!("Pinging {:?} on {:?}", public_key, ip);
-                        let _ = c
+                        let (primary, secondary) = dual_target.get_targets()?;
+                        telio_log_debug!(
+                            "Pinging primary target {:?} on {:?}",
+                            public_key,
+                            primary
+                        );
+
+                        if let Err(e) = c
                             .pinger_client
-                            .pinger(IpAddr::V4(ip4), PingIdentifier(112))
+                            .pinger(primary, PingIdentifier(112))
                             .await
                             .send_ping(PingSequence(211), &[0; PING_PAYLOAD_SIZE])
-                            .await?;
+                            .await
+                        {
+                            telio_log_warn!("Primary target failed: {}", e.to_string());
```

Comment:
When will this actually fail to send to fall back to secondary target?

Reply:
This can fail on a socket level, e.g. OS has IPv6 support turned off.